### PR TITLE
ILB-6  added the functionality edit details of vendor

### PR DESCRIPTION
**Pull request by Sree0076**
added the functionality edit details vendor  and also disabled the vendor name input  during edit
image

### CODE REVIEW

File: src/app/features/vendor/models/vendor.model.ts [hunk 1]
```diff
@@ -28,3 +28,14 @@ export interface IDropDownFields {
   name: string;
 }
 
+export interface IVendorData{
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/models/vendor.model.ts]: IVendorData and IVendorCreation interfaces are almost the same. Create an interface IVendorBasicInfo that they both extend. There is no need to export this interface if it is not required.

File: src/app/features/vendor/services/vendor.service.ts [hunk 2]
```diff
@@ -26,4 +26,14 @@ export class VendorService {
   addVendor(vendorData: IVendorCreation): Observable<IVendorCreation> {
     return this.http.post<IVendorCreation>(`${this.baseUrl}/Vendor/CreateVendor`, vendorData);
   }
-}
+
+  // update vendor data to the backend
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/services/vendor.service.ts]: update the method description

File: src/app/features/vendor/vendor-creation/vendor-creation.component.html [hunk 3]
```diff
@@ -35,6 +36,7 @@ <h2>New vendor</h2>
               <p-dropdown
                 formControlName="state"
                 [options]="states"
+
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.html]: Remove unwanted line

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 4]
```diff
@@ -37,25 +52,39 @@ import { VendorService } from '../services/vendor.service';
...
+  this.fetchMarkets();
+  this.fetchServices();
+  this.createAddVendorForm();
+  if (this.isEdit) {
+    this.selectedVendorId = Number(history.state.id);
+    this.fetchVendorData();
+  }
 }
 
+
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: remove this unwanted space

Comment (Sree0076) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: removed

File: src/app/features/vendor/vendor-creation/vendor-creation.component.html (hunk 3 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.html]: ok

File: src/app/features/vendor/services/vendor.service.ts (hunk 2 above)
Comment (Sree0076) [reply src/app/features/vendor/services/vendor.service.ts]: done

File: src/app/features/vendor/models/vendor.model.ts (hunk 1 above)
Comment (Sree0076) [reply src/app/features/vendor/models/vendor.model.ts]: ok

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 5]
```diff
@@ -70,27 +99,59 @@ ngOnInit(){
...
+        this.vendorData = data;
+        this.populateForm();
+      });
+  }
+}
+
+/**
+* populate the form with retrieved deatails of vendor.
+*/
+populateForm(): void {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: update method name to bindVendorDetails

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 6]
```diff
@@ -101,8 +162,16 @@ fetchServices(): void {
     website: formValue.website,
     serviceId: formValue.service,
     isApproved: false,
-    marketIds: formValue.markets
+    marketIds: formValue.markets,
   };
-  this.vendorService.addVendor(vendorData).subscribe();
+  console.log(vendorData);
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: remove console.log

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 7]
```diff
@@ -101,8 +162,16 @@ fetchServices(): void {
...
+  if (this.isEdit) {
+    this.vendorService.updateVendor(vendorData).subscribe();
+  } else {
+    this.vendorService.addVendor(vendorData).subscribe( );
+  }
 }
 }
+
+
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts:137]: remove unwanted lines

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 8]
```diff
@@ -32,7 +33,7 @@ import { IVendor } from '../models/vendor.model';
 })
 export class VendorListingComponent implements OnInit {
   vendors!: IVendor[];
-
+ constructor(private router : Router){}
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts:36]: add empty line before and after constructor

Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts:36]: Do we using router in this component?

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 5 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 8 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts:36]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 6 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 8 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts:36]: in wiew screen it is using thats why its not removed

File: src/app/features/vendor/vendor-routing.module.ts [hunk 9]
```diff
@@ -6,6 +6,7 @@ import { VendorCreationComponent } from './vendor-creation/vendor-creation.compo
 const routes: Routes = [
   { path: '', component: VendorListingComponent },
   { path: 'creation', component: VendorCreationComponent },
+  { path: 'edit', component: VendorCreationComponent },
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-routing.module.ts]: Add vendor id in path to specify the selected vendor.
eg: edit/id

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 8 above)
Comment (kavyamolr) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts:36]: Remove unused imports.

File: src/app/features/vendor/models/vendor.model.ts [hunk 10]
```diff
@@ -28,3 +18,12 @@ export interface IDropDownFields {
   name: string;
 }
 
+export interface IVendorData extends ICommonVendorFields {
+  service: IDropDownFields;
+  markets: IDropDownFields[];
+}
+export interface IVendor extends ICommonVendorFields,IVendorData {
```
Comment (kavyamolr) [inline src/app/features/vendor/models/vendor.model.ts]: Add empty line between interfaces and remove unwanted lines

Comment (kavyamolr) [inline src/app/features/vendor/models/vendor.model.ts]: IVendorData already have commonvendorfields right? Do we need to extend again?

File: src/app/features/vendor/vendor-routing.module.ts (hunk 9 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-routing.module.ts]: done

**Review by kavyamolr — APPROVED** [review]

**Pull request merged**